exporters/metric/dynamicconfig/servicereader.go

```diff
@@ -0,0 +1,95 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dynamicconfig
+
+import (
+	"context"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+
+	resourcepb "github.com/open-telemetry/opentelemetry-proto/gen/go/resource/v1"
+	pb "github.com/vmingchen/opentelemetry-proto/gen/go/collector/dynamicconfig/v1"
+)
+
+type ServiceReader struct {
+	// Required
+	configHost string
+
+	// Timestamp of last time config service was checked
+	lastTimestamp time.Time
+
+	// Most recent config version
+	lastKnownFingerprint []byte
+
+	// Suggested time to wait before checking config service again (seconds)
+	suggestedWaitTimeSec int32
+
+	// Required. Label to identify this instance
+	resource *resourcepb.Resource
+}
+
+func newServiceReader(configHost string, resource *resourcepb.Resource) *ServiceReader {
+	return &ServiceReader{
+		configHost: configHost,
+		resource:   resource,
+	}
+}
+
+// Reads from a config service
+func (r *ServiceReader) readConfig() (*Config, error) {
+	// Wait for the suggestedWaitTimeSec
+	if !r.lastTimestamp.IsZero() && r.suggestedWaitTimeSec != 0 {
+		suggestedReadTime := r.lastTimestamp.Add(time.Duration(r.suggestedWaitTimeSec) * time.Second)
```

Comment:
simply "r.suggestedWaitTimeSec * time.Second"

Reply:
I ran into problems when I tried to do so before. I think it needs to be a time.Duration * time.Duration, and that a integer literal will get translated into time.Duration automatically, but a variable integer won't.

This link explains it: https://stackoverflow.com/questions/17573190/how-to-multiply-duration-by-integer